fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java

```diff
@@ -581,6 +581,9 @@ public void readFields(DataInput in) throws IOException {
             if (transactionState.isExpired(now)) {
                 LOG.info("discard expired transaction state: {}", transactionState);
                 continue;
+            } else if (transactionState.getTransactionStatus() == TransactionStatus.UNKNOWN) {
```

Comment:
You should also ignore the UNKNOWN STATUS txn in the DatabaseTransactionMgr.unprotectUpsertTransactionState